YouTube.js

```diff
@@ -26,97 +26,81 @@ function detectWeb(doc, url) {
 function getSearchResults(doc, checkOnly) {
 	var container = doc.getElementById('results') || doc.getElementById('browse-items-primary');
 	if (!container) return false;
-	
+
 	var links = container.getElementsByClassName('yt-uix-tile-link');
 
-	var items = {}, found = false;
-	for (var i=0, n=links.length; i<n; i++) {
+	var items = {},
+		found = false;
+	for (var i = 0, n = links.length; i < n; i++) {
 		var title = ZU.trimInternal(links[i].textContent);
-		var id = getVideoId(links[i].href);
-		if (!title || !id) continue;
-		
+		var link = links[i].href;
+		if (!title || !link) continue;
+
 		if (checkOnly) return true;
-		
+
 		found = true
-		items[id] = title;
+		items[link] = title;
 	}
-	
 	return found ? items : false;
 }
 
-function getVideoId(url) {
-	var id = url.match(/\/watch\?(?:.*)\bv=([0-9a-zA-Z_-]+)/);
-	if (id) return id[1];
-	return false;
-}
-
 function doWeb(doc, url) {
 	if (detectWeb(doc, url) != 'multiple') {
-		getData([getVideoId(url)]);
+		scrape(doc, url);
 	} else {
-		Zotero.selectItems(getSearchResults(doc), function (items) {
+		Zotero.selectItems(getSearchResults(doc), function(items) {
 			if (!items) return true;
 
 			var ids = [];
 			for (var i in items) {
 				ids.push(i);
 			}
-			getData(ids);
+			ZU.processDocuments(ids, scrape);
 		});
 	}
 }
 
-function getData(ids){
-	var uris = ids.map(function(id) { return "//gdata.youtube.com/feeds/videos/" + id; });
-	
-	ZU.doGet(uris, function(text) {	
-		var ns = {"default":"http://www.w3.org/2005/Atom", "media":"http://search.yahoo.com/mrss/", "yt":"http://gdata.youtube.com/schemas/2007"};
-		
-		var parser = new DOMParser();
-		var doc = parser.parseFromString(text, "text/xml");
+function scrape(doc, url) {
+	var newItem = new Zotero.Item("videoRecording");
+	//grab the JSON in the header of the page and remove JS code
+	var data = ZU.xpathText(doc, '//script[contains(text(), "ytplayer.config")]');
+	data = data.match(/ytplayer\.config\s*=(.+);\s*ytplayer\.load/)[1];
```

Comment:
nit: Just in case can you make that `(.+?)`. The xpath can grab multiple script elements in theory and there's some chance that ytplayer.load could be mentioned somewhere else, so lazy operator is a bit safer.